crates/fadroma/ensemble/staking.rs

```diff
@@ -0,0 +1,433 @@
+use std::collections::HashMap;
+
+use crate::prelude::*;
+use super::response::StakingResponse;
+
+#[derive(Clone, Default, Debug)]
+pub(crate) struct DelegationWithUnbonding {
+    delegator: HumanAddr,
+    validator: HumanAddr,
+    amount: Coin,
+    unbonding_amount: Coin,
+    can_redelegate: Coin,
+    accumulated_rewards: Coin,
+}
```

Comment:
Just a suggestion:
Maybe declare this struct's fields as:

```Rust
pub(crate) struct DelegationWithUnbonding {
    delegation: FullDelegation,
    unbonding_amount: Coin
}
```
to make the relationship more clear?

Reply:
This does make sense, but it changes a reference to a delegation's amount for example from `delegation.amount` to `delegation.delegation.amount`... Its a small thing but I'm wondering if it would be better not to change it?